src/lib/api/api.go

```diff
@@ -34,7 +34,7 @@ const (
 // API type for implementing the API interface.
 type API struct {
 	ResponseStatus    string
```

Comment:
In the same way, responseStatus will also have race condition. Kindly change that too just like data and pass that into a.send() func. that should be the final update to this PR before lgtm.